File: extensions/copilot/test/scenarios/test-explain/classes.rb
# Copyright (c) Microsoft Corporation and GitHub. All rights reserved.

class MyClass
    def hello
      puts "Hello, world!"
    end
  end

class MySecondClass
    def greet(name)
        puts "Hello, #{name}!"
    end
end

my_instance = MyClass.new
second_instance = MySecondClass.new
second_instance.greet("Alice")

File: extensions/copilot/test/scenarios/test-explain/classes.py
# Copyright (c) Microsoft Corporation and GitHub. All rights reserved.

class Person:
    def __init__(self, name, age):
        self.name = name
        self.age = age

    def introduce(self):
        return f"My name is {self.name} and I am {self.age} years old."

def create_person(name, age):
    return Person(name, age)

class OtherPerson:
    def __init__(self, name, age):
        self.name = name
        self.age = age